apps/web/app/(app)/gardens/page.tsx

```diff
@@ -1,68 +1,40 @@
-import React from "react";
+"use client";
+
+import React, { useMemo } from "react";
 import Image from "next/image";
 import { clouds1, clouds2, gardenHeader } from "@/assets";
 import { GardenCard } from "@/components";
 import {
-  getTokenGardensDocument,
   getTokenGardensQuery,
+  getTokenGardensDocument,
 } from "#/subgraph/.graphclient";
-import { initUrqlClient, queryByChain } from "@/providers/urql";
-import {
-  localhost,
-  arbitrumSepolia,
-  optimismSepolia,
-  sepolia,
-} from "viem/chains";
-import { getContractsAddrByChain, isProd } from "@/constants/contracts";
+import useSubgraphQueryMultiChain from "@/hooks/useSubgraphQueryMultiChain";
+import LoadingSpinner from "@/components/LoadingSpinner";
 
 export const dynamic = "force-dynamic";
 
-const { urqlClient } = initUrqlClient();
-
-export default async function Gardens() {
-  const chainsId = [
-    localhost.id,
-    arbitrumSepolia.id,
-    optimismSepolia.id,
-    sepolia.id,
-  ];
-  let gardens: getTokenGardensQuery | null = null;
-  gardens = {
-    tokenGardens: [],
-  };
+export default function Gardens() {
+  const { data: gardens, fetching } =
+    useSubgraphQueryMultiChain<getTokenGardensQuery>(
+      getTokenGardensDocument,
+      {},
+      {},
+      [
+        {
+          topic: "garden",
+        },
+        {
+          topic: "community",
+        },
+      ],
+    );
 
-  try {
-    if (isProd) {
-      const r0 = await getTokenGardens(sepolia.id);
-      gardens?.tokenGardens.push(...r0.data.tokenGardens);
-      const r1 = await getTokenGardens(arbitrumSepolia.id);
-      gardens?.tokenGardens.push(...r1.data.tokenGardens);
-    } else {
-      const promises = [];
-      for (let index = 0; index < chainsId.length; index++) {
-        const chainId = chainsId[index];
-        const subgraph = getContractsAddrByChain(chainId)?.subgraphUrl;
+  const tokenGardens = useMemo(() => {
+    return gardens
+      ?.flatMap((g) => g.tokenGardens)
+      .filter((x): x is NonNullable<typeof x> => !!x);
+  }, [gardens]);
```

Comment:
Not clear what its the idea here, remove null? still can return undefined right?

Flat functions sometimes replace duplicated stuffs, its safe here?

Reply:
I was trying to have the same logic as before but without for loops with push on demands on an array since the code is not sync anymore. (Cause Client side) The idea is to take all gardens from each chain flat them in a single level array and filter out the gardens that are undefined. Dont know why sone can be but there was the same check as before

Reply:
Do you have suggestions ?

Reply:
NotNullable maybe dont check for undefined, but should, maybe doing simple check for undefined and null should return the correct type

Reply:
I will check that

Reply:
@Corantin marking as solved